main.go

```diff
@@ -221,7 +222,13 @@ func handlePost(res http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	now := time.Now()
 	date, _ := time.ParseInLocation("2006-01-02", dateStr, loc)
+	date = date.Add(20*time.Hour)
+	if now.After(date) && date != locatedNullDate {
```

Comment:
Use [date.IsZero()](https://godoc.org/time#Time.IsZero) instead of locatedNullDate (also, that variable name is wrong, but that doesn't matter, as it's going away anyway).